util/src/org/commcare/util/mocks/MockDataUtils.java

```diff
@@ -41,15 +41,32 @@ public static MockUserDataSandbox getStaticStorage() {
         return new MockUserDataSandbox(factory);
     }
 
-    public static void parseIntoSandbox(InputStream stream, MockUserDataSandbox sandbox) {
-        CommCareTransactionParserFactory factory = new CommCareTransactionParserFactory(sandbox);
-        try {
-            DataModelPullParser parser = new DataModelPullParser(stream, factory);
-            parser.parse();
-        } catch (IOException | UnfullfilledRequirementsException |
-                XmlPullParserException | InvalidStructureException ioe) {
-            ioe.printStackTrace();
+    /**
+     * Parse the transactions int he provided stream into the user sandbox provided.
+     *
+     * Will rethrow any errors and failfast (IE: Parsing will stop)
+     */
+    public static void parseIntoSandbox(InputStream stream, MockUserDataSandbox sandbox) throws IOException,
+            UnfullfilledRequirementsException,
+            XmlPullParserException, InvalidStructureException {
+        parseIntoSandbox(stream, sandbox, true);
+    }
+
+    /**
+     * Parse the transactions int he provided stream into the user sandbox provided.
+     *
+     * If failfast is true, will rethrow any errors and failfast (IE: Parsing will stop)
+     */
+    public static void parseIntoSandbox(InputStream stream, MockUserDataSandbox sandbox, boolean failfast) throws IOException,
+            UnfullfilledRequirementsException,
+            XmlPullParserException, InvalidStructureException {
+        if(stream == null) {
+            throw new IOException("Parse Stream is Null");
         }
+
+        CommCareTransactionParserFactory factory = new CommCareTransactionParserFactory(sandbox);
+        DataModelPullParser parser = new DataModelPullParser(stream, factory, true, failfast);
```

Comment:
The DataModelPullParser constructor being called has the following declaration:
```DataModelPullParser(InputStream is, TransactionParserFactory factory, boolean failfast, boolean deep)``` https://github.com/dimagi/commcare/blob/master/backend/src/org/commcare/data/xml/DataModelPullParser.java#L51

I'm confused by the fact that the 4th argument being passed into the constructor, called  ```failfast```, is getting named to ```deep``` in the body of the constructor, while the 3rd parameter of the constructor is also named ```failfast```.

Are you sure you didn't mean to name the ```failfast``` variable in the ```parseIntoSandbox``` something else, like ```deep```?